machineconfiguration/v1/types_machineosconfig.go

```diff
@@ -0,0 +1,240 @@
+
+	// status describes the status of the machineosconfig
+	// +optional
+	Status *MachineOSConfigStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineOSConfigList describes all configurations for image builds on the system
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type MachineOSConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []MachineOSConfig `json:"items"`
+}
+
+// MachineOSConfigSpec describes user-configurable options as well as information about a build process.
+type MachineOSConfigSpec struct {
+	// machineConfigPool is the pool which the build is for
+	// +kubebuilder:validation:Required
+	MachineConfigPool MachineConfigPoolReference `json:"machineConfigPool"`
+	// buildInputs is where user input options for the build live
+	// +kubebuilder:validation:Required
+	BuildInputs BuildInputs `json:"buildInputs"`
+	// buildOutputs holds all information needed to handle booting the image after a build
+	// +optional
+	BuildOutputs *BuildOutputs `json:"buildOutputs,omitempty"`
+}
+
+// MachineOSConfigStatus describes the status this config object and relates it to the builds associated with this MachineOSConfig
+type MachineOSConfigStatus struct {
+	// conditions are state related conditions for the config.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	// observedGeneration represents the generation observed by the Machine Config Operator's build controller.
+	// +kubebuilder:validation:Required
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+	// currentImagePullspec is the fully qualified image pull spec used by the MCO to pull down the new OSImage. This must include sha256.
+	// The format of the image pullspec is:
+	// host[:port][/namespace]/name@sha256:<digest>
+	// The digest must be 64 characters long, and consist only of lowercase hexadecimal characters, a-f and 0-9.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=447
+	// +kubebuilder:validation:XValidation:rule=`(self.split('@').size() == 2 && self.split('@')[1].matches('^sha256:[a-f0-9]{64}$'))`,message="the OCI Image reference must end with a valid '@sha256:<digest>' suffix, where '<digest>' is 64 characters long"
+	// +kubebuilder:validation:XValidation:rule=`(self.split('@')[0].matches('^([a-zA-Z0-9-]+\\.)+[a-zA-Z0-9-]+(:[0-9]{2,5})?/([a-zA-Z0-9-_]{0,61}/)?[a-zA-Z0-9-_.]*?$'))`,message="the OCI Image name should follow the host[:port][/namespace]/name format, resembling a valid URL without the scheme"
+	// +optional
+	CurrentImagePullspec string `json:"currentImagePullspec,omitempty"`
```

Comment:
I feel like it might be useful to have a reference to the currently active MachineOSBuild here. In practice, I've been using an annotation for this purpose, but I think having it on the status would be better.

Reply:
Added :+1: